Tests.py

```diff
@@ -0,0 +1,150 @@
+from unittest.mock import Mock
```

Comment:
When running your tests they are failing and only 4 are running because of an exception.